recipes/agent.rb

```diff
@@ -100,7 +100,7 @@
 
 # Client utilities for pulp content manager
 package "python3-pip"
-%w[pulp-rpm-client==3.8.0 pulpcore-client==3.8.1].each do |pulppkg|
+%w[pulp-rpm-client==3.4.1 pulpcore-client==3.4.1].each do |pulppkg|
```

Comment:
I did some testing, and it looks like the issues we were having with the newer Pulp versions were caused by a mismatch, and not necessarily our use of the API. Specifically, things seem to go awry if the client code is newer than the API server.

I found that `pulpcore==3.9.1` and `pulp-rpm==3.9.0` (and their corresponding `-client` packages) seemed to work well in concert.

It would probably be worth upgrading everything prior to merging this and capturing the new AMI for the workers.

Reply:
By upgrading everything do you mean updating the pulp version both on the agents and in the pulp image?

Reply:
Correct